server/src/main/java/org/spine3/server/entity/storage/ColumnTypeRegistry.java

```diff
@@ -0,0 +1,156 @@
+ * {@linkplain org.spine3.server.storage.Storage Storage}) call {@link #get(Column)}
+ * on the column which you'd like to retrieve the {@linkplain ColumnType type} for.
+ *
+ * <p>Note, that in the example above, if you try to get the {@linkplain ColumnType type} for
+ * a {@link Column} of e.g. class {@code java.sql.Timestamp}, which {@code extends Date} unless you
+ * specify a {@link ColumnType} for {@code java.sql.Timestamp} explicitly, the same value as for
+ * the class {@code Date} will be returned. I.e. the {@link ColumnType} of a parent class can be
+ * used for a derived class. But be careful, the {@code interface}s are not supported within this
+ * feature.
+ *
+ * <p>If several {@linkplain ColumnType ColumnTypes} are written under a single {@linkplain Class}
+ * object, the latest {@code put} overrides all the previous ones.
+ *
+ * @param <C> storage-specific implementation of the {@link ColumnType}
+ * @author Dmytro Dashenkov
+ */
+public final class ColumnTypeRegistry<C extends ColumnType> {
+
+    private final ImmutableMap<Class, C> columnTypeMap;
+
+    private ColumnTypeRegistry(Builder<C> builder) {
+        this.columnTypeMap = ImmutableMap.copyOf(builder.columnTypeMap);
+    }
+
+    /**
+     * Retrieves the {@link ColumnType} for specified {@link Column}.
+     *
+     * <p>By default, this method returns the {@link ColumnType} for the
+     * {@linkplain Column column's} {@linkplain Column#getType() type}.
+     *
+     * <p>If the {@link ColumnType} was not found by the {@code class} of the {@linkplain Column},
+     * its superclasses are checked one by one until a {@link ColumnType} is found or until
+     * the look up reaches the class representing {@link Object}.
+     *
+     * <p>If no {@link ColumnType} is found, an {@link IllegalStateException} is thrown.
+     *
+     * @param field the {@link Column} to get type conversion strategy for
+     * @return the {@link ColumnType} for the given {@link Column}
+     */
+    public C get(Column<?> field) {
+        checkNotNull(field);
+
+        Class javaType = field.getType();
+        javaType = Primitives.wrap(javaType);
+        C type = null;
+
+        while (type == null && javaType != null) {
+            type = columnTypeMap.get(javaType);
+            javaType = javaType.getSuperclass();
+        }
+
+        checkState(type != null,
+                   "Could not find storage type for %s.",
+                   field.getType());
+        return type;
+    }
+
+    @VisibleForTesting
+    Map<Class, C> getColumnTypeMap() {
+        return Collections.unmodifiableMap(columnTypeMap);
+    }
+
+    public static <C extends ColumnType> Builder<C> newBuilder() {
+        return new Builder<>();
+    }
+
+    public static <C extends ColumnType> Builder<C> newBuilder(ColumnTypeRegistry<C> src) {
+        final Builder<C> builder = new Builder<>();
+        builder.columnTypeMap.putAll(src.columnTypeMap);
```

Comment:
Let's use `Builder.put` calls instead to reuse the validation.

Even if the validation code is expanded with some more sophisticated checks, we'll always be okay.